src/server/routes/weatherLocation.js

```diff
@@ -0,0 +1,219 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+  * License, v. 2.0. If a copy of the MPL was not distributed with this
+  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const express = require('express');
+const { log } = require('../log');
+const { getConnection } = require('../db');
+const { adminAuthMiddleware, optionalAuthMiddleware } = require('./authenticator');
+const WeatherLocation = require('../models/WeatherLocation');
+const moment = require('moment');
+const Reading = require('../models/Reading');
+const validate = require('jsonschema').validate;
+const Point = require('../models/Point');
+const { success, failure } = require('./response');
+const WeatherData = require('../models/WeatherData');
+const { fetchWeatherData } = require('../services/weather/fetchData')
+
+const router = express.Router();
+
+function formatWeatherLocationForResponse(item) {
+	return {
+		id: item.id,
+		identifier: item.identifier,
+		gps: item.gps,
+		note: item.note
+	};
+}
+
+/**
+ * Route for getting all weather locations.
+ */
+router.get('/', optionalAuthMiddleware, async (req, res) => {
+	const conn = getConnection();
+	try {
+		const rows = await WeatherLocation.getAll(conn);
+		res.json(rows.map(formatWeatherLocationForResponse));
+	} catch (err) {
+		log.error(`Error while performing GET weather location details query: ${err}`, err);
+		failure(res, 500, `Error while retrieving weather locations ${err}`);
+	}
+});
+
+// This checks params for both edit and create. In principle they could differ since not all needed for create
+// but they are the same due to current routing for now.
+function validateWeatherLocationParams(params) {
+	const validParams = {
+		type: 'object',
+		required: ['identifier', 'gps'],
+		properties: {
+			// Removed id from properties list since it is set to undefined no matter what is passed.
+			identifier: {
+				type: 'string',
+				minLength: 1
+			},
+			gps: {
+				oneOf: [
+					{
+						type: 'object',
+						required: ['latitude', 'longitude'],
+						properties: {
+							latitude: { type: 'number', minimum: '-90', maximum: '90' },
+							longitude: { type: 'number', minimum: '-180', maximum: '180' }
+						}
+					},
+					{ type: 'null' }
+				]
+			},
+			note: {
+				oneOf: [
+					{ type: 'string' },
+					{ type: 'null' }
+				]
+			}
+		}
+	}
+	const paramsValidationResult = validate(params, validParams);
+	return { valid: paramsValidationResult.valid, errors: paramsValidationResult.errors };
+}
+
+/**
+ * Fetches and inserts weather data for a newly created weather location.
+ * If the location has no existing weather data, it retrieves data starting from the earliest reading timestamp.
+ * @param location - Instance of WeatherLocation.
+ * @param conn - The database connection.
+ */
+async function addWeatherDataForLocation(location, conn) {
```

Comment:
When OED gets meter data it logs the start and then the end for the record. I think the same should happen here.